octodns/zone.py

```diff
@@ -68,10 +68,10 @@ def add_record(self, record, replace=False, lenient=False):
             self.hydrate()
 
         name = record.name
-        last = name.split('.')[-1]
 
-        if not lenient and last in self.sub_zones:
-            if name != last:
+        if not lenient and any(map(lambda sz: name.endswith(sz),
+                                   self.sub_zones)):
```

Comment:
Again if my understanding is correct something like may work:

```suggestion
        if not lenient and any([name.endswith(sz) for sz in self.sub_zones]):
```